src/js/components/forms/TextInput.jsx

```diff
@@ -1,12 +1,35 @@
 import React, { PropTypes } from 'react';
+import { Icon } from '@jenkins-cd/react-material-icons';
+
 import { TextControl } from './TextControl';
 
+
+// wraps the Icon in a div to provide consistent cursor behavior
+function NestedIcon(props) {
+    return (
+        <div className={`TextInput-icon ${props.className}`}>
+            <Icon icon={props.icon} />
+        </div>
+    );
+}
+
+NestedIcon.propTypes = {
+    className: PropTypes.string,
+    icon: PropTypes.string,
+};
+
+
 export class TextInput extends React.Component {
 
     render() {
+        const classLeft = this.props.iconLeft ? 'u-icon-left' : '';
+        const classRight = this.props.iconRight ? 'u-icon-right': '';
+
         return (
-            <TextControl {...this.props} className={`TextInput ${this.props.className}`}>
+            <TextControl {...this.props} className={`TextInput ${this.props.className} ${classLeft} ${classRight}`}>
+                { classLeft && <NestedIcon className={classLeft} icon={this.props.iconLeft} /> }
```

Comment:
I'll certainly add that in once we have a use case for it. Right now I'm supporting "icons as decoration" only.